services/crates/crates-msrv.service.js

```diff
@@ -0,0 +1,78 @@
+import { NotFound, pathParams } from '../index.js'
+import {
+  BaseCratesService,
+  description as cratesIoDescription,
+} from './crates-base.js'
+
+const description = `
+${cratesIoDescription}
+
+MSRV is a crate's minimum suppported rust version,
+the oldest version of Rust supported by the crate.
+See the [Cargo Book](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field)
+for more info.
+`
+
+export default class CratesMSRV extends BaseCratesService {
+  static category = 'platform-support'
+  static route = {
+    base: 'crates/msrv',
+    pattern: ':crate/:version?',
+  }
+
+  static openApi = {
+    '/crates/msrv/{crate}': {
+      get: {
+        summary: 'Crates.io MSRV',
+        description,
+        parameters: pathParams({
+          name: 'crate',
+          example: 'serde',
+        }),
+      },
+    },
+    '/crates/msrv/{crate}/{version}': {
+      get: {
+        summary: 'Crates.io MSRV (version)',
+        description,
+        parameters: pathParams(
+          {
+            name: 'crate',
+            example: 'serde',
+          },
+          {
+            name: 'version',
+            example: '1.0.194',
+          },
+        ),
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'MSRV', color: 'blue' }
+
+  static render({ msrv: message }) {
+    return { message }
+  }
+
+  static transform({ errors, version, versions }) {
+    // crates.io returns a 200 response with an errors object in
+    // error scenarios, e.g. https://crates.io/api/v1/crates/libc/0.1
+    if (errors) {
+      throw new NotFound({ prettyMessage: errors[0].detail })
+    }
```

Comment:
I'm not going to insist we do it to merge this, but I'm making a note to myself here: We should move this logic up to the base class because it affects all services that call this API.